src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -119,6 +119,29 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 		}
 		if (!this._table) {
 			let source = <IDataResource><any>this._bundleOptions.data[this.mimeType];
+			let columnNames: Array<string> = [];
+			let rowIndex = 0;
+			// Get schema list of grid
+			for (let i of source.schema.fields) {
+				columnNames.push(i.name);
+			}
+			// Check to see if data source is ordered properly based on schema
+			// SQL notebooks do not use columnName as key (instead uses indices)
+			// So we add a condition for SQL notebooks to not be reordered
+			if (source.data.length > 0) {
+				if (columnNames !== Object.keys(source.data[0]) && Object.keys(source.data[0])[0] !== '0') {
```

Comment:
!== is doing direct comparison, so `columnNames !== Object.keys(...)` is always going to be true because they aren't the same array. You should use the equals function https://github.com/microsoft/azuredatastudio/blob/main/src%2Fvs%2Fbase%2Fcommon%2Farrays.ts#L27

Reply:
Actually, I found that the condition won't always be true. The input notebook that I show on the left in my image skips this entire reordering since it doesn't pass the condition (meaning that the columnNames and the keys of the first row are equal). I only want to reorder the table when they do not match, utilizing the equals function would return true for the above scenario wasting time to go through a reordering that would not be needed. I can still utilize the function but with a **!**equals(..).

Reply:
Not sure what you mean - the check as it is is guaranteed to be true. Two arrays will never be strictly equal (===) unless they are the same object, and since you're using map here that's never going to be able to be true

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgJwQRhgXhgbWQGhgJjwGYBdAKFElkQX3S1wOPMohABsBTAOnZAHMAFDVRox8JPgCUQA

Unless you're saying the intent of the logic (checking that the values are in the same order) won't always be true. Which is fine and I understood. I was just saying that the logic isn't doing what I assume you meant to do. 

And yes of course, I meant use the equals function to do the comparison - not that you would do this logic if they're equal 😄

Reply:
The intent of the logic which I may have not been clear was to just check if the values are in the same order. I agree that the entities and two arrays won't ever be strictly equal. I think I misunderstood what you initially stated as comparing the values strictly, when in fact I just want to make sure they are in the same order. Thanks!